src/Confluent.SchemaRegistry.Serdes.Protobuf/ProtobufUtils.cs

```diff
@@ -298,42 +299,36 @@ private static ISet<string> GetInlineTags(FieldDescriptorProto fd)
             return tags;
         }
 
-        public static FileDescriptorSet Parse(string schema, IDictionary<string, string> imports)
+        internal static FileDescriptorSet Parse(string schema, IDictionary<string, string> imports)
         {
-            IDictionary<string, string> allImports = new Dictionary<string, string>(BuiltIns);
-            imports?.ToList().ForEach(x => allImports.Add(x.Key, x.Value));
-            
-            var fds = new FileDescriptorSet();
-            fds.FileSystem = new ProtobufImports(allImports);
-            
-            fds.Add("__root.proto", true, new StringReader(schema));
-            foreach (KeyValuePair<string, string> import in allImports)
+            var fds = new FileDescriptorSet
             {
-                fds.AddImportPath(import.Key);
-                
-            }
+                FileSystem = new ProtobufImports(imports)
+            };
+            fds.Add("__root.proto", true, new StringReader(schema));
+            fds.AddImportPath(""); // all imports are relative in the filesystem so must make import path just empty string
             fds.Process();
             return fds;
-        } 
-        
-        class ProtobufImports : IFileSystem
+        }
+
+        private class ProtobufImports : IFileSystem
         {
-            protected IDictionary<string, string> Imports { get; set; }
+            private readonly IDictionary<string, string> imports;
 
             public ProtobufImports(IDictionary<string, string> imports)
             {
-                Imports = imports;
+                this.imports = imports;
             }
 
             public bool Exists(string path)
             {
-                return Imports.ContainsKey(path);
+                return BuiltIns.ContainsKey(path) || (imports?.ContainsKey(path) ?? false);
```

Comment:
By using BuiltIns here, I avoid the creation of the additional dictionary of all the types with every call to the Parse function.